python/pyhail/context.py

```diff
@@ -0,0 +1,457 @@
+import pyspark
+
+from pyhail.dataset import VariantDataset
+from pyhail.java import jarray, scala_object
+
+class HailContext:
+    """:class:`.HailContext` is the main entrypoint for PyHail
+    functionality.
+    
+    :param SparkContext sc: The pyspark context.
+    """
+    
+    def __init__(self, sc):
+        self.sc = sc
+        
+        self.gateway = sc._gateway
```

Comment:
Do we expect that the users will ever be manipulating the java objects?  We should put a leading underscore before everything we intend to be private, in Python convention.